app/Providers/AuthServiceProvider.php

```diff
@@ -43,16 +44,28 @@ public function boot(): void
     private function defineGates(): void
     {
         Gate::define('view-project', function (?User $user, Project $project) {
-            return ProjectPermissions::canViewProject($project, $user);
+            if (!ProjectPermissions::canViewProject($project, $user)) {
+                return Response::denyAsNotFound('You do not have access to the requested project or the requested project does not exist.');
```

Comment:
When attempting to access a private project as an unprivileged user, the page doesn't respond the way that I expect it to: It shows a blank screen with no indication what the error is visible on the page.  I had to open the console to see what the response from the page load was: 

![image](https://github.com/Kitware/CDash/assets/542106/e9646c0c-b624-437b-9009-2b1c3b808d8f)

Is this situation because this uses `denyAsNotFound` instead of `denyWithStatus`?

Reply:
I don't think it's a problem with `denyAsNotFound` vs `denyWithStatus`.  As per the [documentation](https://laravel.com/docs/10.x/authorization#customising-gate-response-status), `Response::denyAsNotFound()` is just shorthand for `Response::denyWithStatus(404)`.  This problem also exists on master and seems to be an issue with AngularJS error handling rather than something in this PR.  It is possibly something I broke in #1351--I'll make a separate PR to fix it.